### PR TITLE
Allow rust to cross-compile under scratchbox

### PR DESCRIPTION
**Pull request by lbt**
Add support for default targets
Enable use of host-cc
Partial fix for fork/exec deadlocks in rust's std:: library

### CODE REVIEW

File: rust.spec [hunk 1]
```diff
@@ -25,21 +41,49 @@ URL:            https://www.rust-lang.org
...
+BuildRequires:  cross-armv7hl-binutils
+BuildRequires:  cross-armv7hl-as
+BuildRequires:  cross-armv7hl-glibc
+BuildRequires:  cross-armv7hl-glibc-devel
+BuildRequires:  cross-armv7hl-glibc-headers
+BuildRequires:  cross-armv7hl-kernel-headers
+%if 0%{?build_aarch64}
+BuildRequires:  cross-aarch64-gcc
+BuildRequires:  cross-aarch64-binutils
+#BuildRequires:  cross-aarch64-as
```
Comment (xfade) [inline rust.spec]: Don't we need this?

File: rust.spec [hunk 2]
```diff
@@ -205,6 +267,9 @@ find -name '*.rs' -type f -perm /111 -exec chmod -v -x '{}' '+'
 %build
 
 export RUSTFLAGS="%{rustflags}"
+CFLAGS=
```
Comment (xfade) [inline rust.spec:292]: Why do we reset these?

File: rust.spec [hunk 3]
```diff
@@ -296,6 +396,7 @@ rm -f %{buildroot}%{_bindir}/rustdoc
 rm -fr %{buildroot}%{_mandir}/man1
 
 %check
+%if 0
```
Comment (xfade) [inline rust.spec:433]: Needs to be cleaned up?

File: rust.spec (hunk 2 above)
Comment (lbt) [reply rust.spec:292]: comment added

**Review by xfade — APPROVED** [review]
LGTM

**Review by mkosola — APPROVED** [review]

**Pull request merged**